sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -1461,7 +1465,8 @@ void _handle_search_rtr(gnrc_netif_t *netif)
 {
 #if !IS_ACTIVE(CONFIG_GNRC_IPV6_NIB_NO_RTR_SOL)
     gnrc_netif_acquire(netif);
-    if (!(gnrc_netif_is_rtr_adv(netif)) || gnrc_netif_is_6ln(netif)) {
+    if (!gnrc_netif_is_6lbr(netif) &&
+        (!gnrc_netif_is_rtr_adv(netif) || gnrc_netif_is_6ln(netif))) {
```

Comment:
Since you have written this condition three times now I'd say better move it to a helper function

```C
static inline bool _should_send_rs(gnrc_netif_t *netif)
{
    return !gnrc_netif_is_6lbr(netif) && (!gnrc_netif_is_rtr_adv(netif) || gnrc_netif_is_6ln(netif));
}
```